Client/WebFilters/CosmeticFiltersResourceDownloader.swift

```diff
@@ -69,54 +86,56 @@ class CosmeticFiltersResourceDownloader {
         let datFileUrls = filePaths?.filter { $0.pathExtension == "dat" }
         let jsonFileUrls = filePaths?.filter { $0.pathExtension == "json" }
         
-        datFileUrls?.forEach {
+        try await datFileUrls?.asyncForEach({
             let fileName = $0.deletingPathExtension().lastPathComponent
             if let data = fm.contents(atPath: $0.path) {
-                setDataFile(data: data, id: fileName)
+                try await self.setDataFile(into: engine, data: data, id: fileName)
             }
-        }
+        })
         
-        jsonFileUrls?.forEach {
+        try await jsonFileUrls?.asyncForEach {
             let fileName = $0.deletingPathExtension().lastPathComponent
             if let data = fm.contents(atPath: $0.path) {
-                setJSONFile(data: data, id: fileName)
+                try await self.setJSONFile(into: engine, data: data, id: fileName)
             }
         }
     }
     
-    private func downloadCosmeticSamples() {
-        downloadResources(type: .cosmeticSample,
-                          queueName: "CSS Queue").uponQueue(.main) {
-            log.debug("Downloaded Cosmetic Filters CSS Samples")
+    private func downloadCosmeticSamples(with engine: AdblockRustEngine) async throws {
+        try await downloadResources(for: engine, type: .cosmeticSample, queueName: "CSS-Queue")
+        log.debug("Downloaded Cosmetic Filters CSS Samples")
+        
+        await MainActor.run {
             Preferences.Debug.lastCosmeticFiltersCSSUpdate.value = Date()
         }
     }
     
-    private func downloadResourceSamples() {
-        downloadResources(type: .resourceSample,
-                          queueName: "Scriplets Queue").uponQueue(.main) {
-            log.debug("Downloaded Cosmetic Filters Scriptlets Samples")
+    private func downloadResourceSamples(with engine: AdblockRustEngine) async throws {
```

Comment:
Just mark this as `@MainActor` isolated instead of doing `await MainActor.run` at the end